server/src/main/java/org/opensearch/index/translog/transfer/FileTransferTracker.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.translog.transfer;
+
+import org.opensearch.index.shard.ShardId;
+import org.opensearch.index.translog.transfer.FileSnapshot.TransferFileSnapshot;
+import org.opensearch.index.translog.transfer.listener.FileTransferListener;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+/**
+ * FileTransferTracker keeps track of translog files uploaded to the remote translog
+ */
+public class FileTransferTracker implements FileTransferListener {
```

Comment:
Would concurrent uploads be possible in which case we need to make this class Thread safe

Reply:
Thanks for calling this out. We will only do one `transferSnapshot` upload at a time . This implies that for a given generation only one thread will be updating the Tracker.  So as of now we don't need to make this class Thread safe. Will call it out explicitly though .

Reply:
Do you need the ConcurrentHashMap as opposed to a regular HashMap?